FF1Lib/ListRNGExtensions.cs

```diff
@@ -44,6 +44,31 @@ public static T PickRandomItemWeighted<T>(this IList<(T Item, int Weight)> weigh
 			int randomNumber = rng.Between(1, weightedList.Sum(item => item.Weight));
 			return rangedItems.First(item => randomNumber <= item.RangeTo).Item;
 		}
+
+		[DebuggerStepThrough]
+		public static T PickRemoveRandomItemWeighted<T>(this IList<(T Item, int Weight)> weightedList, MT19337 rng)

```

Comment:
Rename to SpliceRandomItemWeighted for consistency

Reply:
Updated for that name, fixed a few minor issues.